src/WhoWroteThat.ts

```diff
@@ -47,11 +47,49 @@ export default class WhoWroteThat {
                 const declaration =
                     this.parser.findDeclarationByLine(rootNode, line);
                 return this.rec(declaration, commits, 0)
-                    .then(owners => ({ declaration, owners }));
+                    .then(owners => ({declaration, owners}));
             }).then(transformResult);
         });
     }
 
+    file(): Promise<CodeOwners> {
+        return Git.commitsAffectingFile(this.filePath).then(commits => {
+            return readFile(this.filePath).then(sourceCode => {
+                const rootNode = this.parser.parse(sourceCode);
+                const declaration =
+                    this.parser.findSourceFileDeclaration(rootNode);
+                declaration.name = this.filePath.split('/').pop();
```

Comment:
IMHO it would be nicer to pass this value as an argument to `findSourceFileDeclaration` and set the `name` member there :)

Reply:
i was thinking about that too - but i think `Parser` has nothing to do with filepaths.
maybe its a good idea to move the whole function to `WhoWroteThat.ts`